nixos/modules/services/networking/soju.nix

```diff
@@ -122,6 +133,13 @@ in
           must both be specified to enable TLS.
         '';
       }
+      {
+        assertion = !(cfg.configFile != configFile && cfg.extraConfig != "");
+        message = ''
+          services.soju.configFile and services.soju.extraConfig
+          must not be set at the same time.
+        '';
+      }
```

Comment:
This assertion seems a bit weird to me -- why only check for `extraConfig` and none of the other options? Personally I think it would be better to remove the assertion and document that `extraConfig` only applies to the generated config file.